sdc/tests/test_dataframe.py

```diff
@@ -1077,6 +1078,117 @@ def test_impl(df, n):
                 with self.subTest(n=n, index=idx):
                     pd.testing.assert_frame_equal(sdc_func(df, n), test_impl(df, n))
 
+    def test_df_iloc_slice(self):
+        def test_impl(df, n, k):
+            return df.iloc[n:k]
+        sdc_func = sdc.jit(test_impl)
+        cases_idx = [[3, 4, 2, 6, 1], None]
+        cases_n = [-10, 0, 8, None]
+        for idx in cases_idx:
+            df = pd.DataFrame({"A": [3.2, 4.4, 7.0, 3.3, 1.0],
+                               "B": [5.5, np.nan, 3, 0, 7.7],
+                               "C": [3, 4, 1, 0, 222]}, index=idx)
+            for n in cases_n:
+                for k in cases_n[::-1]:
+                    with self.subTest(index=idx, n=n, k=k):
+                        pd.testing.assert_frame_equal(sdc_func(df, n, k), test_impl(df, n, k))
+
+    def test_df_iloc_values(self):
+        def test_impl(df, n):
+            return df.iloc[n, 1]
+        sdc_func = sdc.jit(test_impl)
+        cases_idx = [[3, 4, 2, 6, 1], None]
+        cases_n = [1, 0, 2]
+        for idx in cases_idx:
+            df = pd.DataFrame({"A": [3.2, 4.4, 7.0, 3.3, 1.0],
+                               "B": [5.5, np.nan, 3, 0, 7.7],
+                               "C": [3, 4, 1, 0, 222]}, index=idx)
+            for n in cases_n:
+                with self.subTest(index=idx, n=n):
+                    if not (np.isnan(sdc_func(df, n)) and np.isnan(test_impl(df, n))):
+                        self.assertEqual(sdc_func(df, n), test_impl(df, n))
+
+    def test_df_iloc_int_value_error(self):
+        def test_impl(df):
+            return df.iloc[11]
+        sdc_func = sdc.jit(test_impl)
+        df = pd.DataFrame({"A": [3.2, 4.4, 7.0, 3.3, 1.0],
+                           "B": [5.5, np.nan, 3, 0, 7.7],
+                           "C": [3, 4, 1, 0, 222]})
+
+        with self.assertRaises(IndexingError) as raises:
+            sdc_func(df)
```

Comment:
I propose to combine all the tests with such error to a single one because of significant common part. I mean just join the code.